backend/api/activity/index.js

```diff
@@ -6,68 +6,224 @@ const activity = express.Router();
 
 /**
  * Creates a new activity and returns its data
+ * @param {Hex} id - The id of the activity, must be only hexadecimal
  * @body {String} title - The title of the post
  * @body {String} description - The title of the post
+ * @body {Integer} startDateTime - The date/time the activity will start in milliseconds
+ * @body {Integer} endDateTime - The date/time the activity will end in milliseconds
  */
 activity.route('/create')
-  .post(passport.authenticate('local'), async (req, res) => {
+  .post(async (req, res) => {
+	//Ensure user is authenticated
     if (!req.user) {
-      return res.status(403).json({
+      return res.status(401).json({
         error: 'Not authenticated',
       });
     }
+	//Create the activity
     const statusCode = 200;
-    const currentDate = Date.now();
+    const currentDateTime = Date.now();
     const activitydata = new Activity({
-      title: req.body.title,
-      description: req.body.description,
-      creationDateTime: currentDate,
-      lastEditDateTime: currentDate,
+      title: req.body.title || "Unnamed activity",
+      description: req.body.description || "No description offered",
+      creationDateTime: currentDateTime,
+      lastEditDateTime: currentDateTime,
+	  startDateTime: req.body.startDateTime || (currentDateTime + 86400000),
+	  endDateTime: req.body.endDateTime || (currentDateTime + 90000000),
       postedBy: req.user.id,
     });
+	
+	//Ensure end time comes after start time
+	if (activitydata.endDateTime < activitydata.startDateTime) {
+	  return res.status(400).json({
+        error: 'endDateTime is less than startDateTime',
+      });
+	}
+	
     try {
+	  //Save it to the database
       await activitydata.save();
+	  
+	  //Response
       return res.status(statusCode).json({ activity: activitydata });
     } catch (error) {
+	  //Error Response
       return res.status(500).send(error);
     }
   });
 
 /**
  * Edits activity by id
  * TODO: Implement this
- * @param {Number} id - The id of the activity, must be only digits
+ * @param {Hex} id - The id of the activity, must be only hex digits
  */
-activity.route('/edit/:id([0-9]+)')
-  .post(passport.authenticate('local'), async (req, res) => {
-    if (!req.user) {
-      return res.status(403).json({
+activity.route('/edit/:id([a-f0-9]+)')
+  .post(async (req, res) => {
+	//Ensure user is authenticated
+	if (!req.user) {
+      return res.status(401).json({
         error: 'Not authenticated',
       });
     }
-    const statusCode = 200;
-    const shortened = Boolean(req.query.shortened);
-    return res.status(statusCode).json({
-      id: req.params.id,
-      status: statusCode,
-      shortened,
-    });
+	
+	try {
+	  //Get the activity to edit
+      const getobj = await Activity.findById(req.params.id).exec();
```

Comment:
We should wrap each method that interacts with the database in its own `try...catch`. The way we have it now, our catch could be handling either this line, or line 98 when saving back to the database.

Reply:
How do you think that should be formatted? If we nest the try/catch blocks, it could look messy. If we put only the database interaction in the try block, the value can't be accessed outside of the scope of the try block.

Reply:
It should be like this:
```js
// Define the variable that holds the found document here in case the scope of the try blocks it from being used below
let getobj;

try {
  getobj = await Activity.findById(req.params.id).exex();
} catch (error) {
  // Handle find document error
}

...
// Other code handling the changing
...

try {
  await getobj.save();
} catch (error) {
  // Handle saving document error
}
```

Here each interaction is safely contained in a `try...catch` and all of the other code that interacts with the document variables is between the two.